Add support to disable/enable roles on the appliance.

fdupont-redhat opened the pull request:
This PR allows to enable or disable role when deploying the appliance. It takes two lists as variables: `miq_disabled_roles` and `miq_enabled_roles`. Both have the same default value `[]` that don't change the list of roles, so the appliance keeps the ManageIQ default roles.

fabiendupont committed:
Add support to disable/enable roles on the appliance.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

ghost commented:
@machacekondra, @mwperina could you please review this PR ?

machacekondra commented on tasks/manage_appliance_roles.yml:
```diff
@@ -0,0 +1,33 @@
+  uri:
+    url: https://{{ miq_ip_addr }}/api/servers/1/settings
+    user: "{{ miq_app_username | default('admin') }}"
+    password: "{{ miq_app_password | default('smartvm') }}"
```
The '| default() ' isn't needed here. and for user.

machacekondra commented on README.md:
```diff
@@ -99,16 +99,18 @@ The item in `miq_vm_nics` list of can contain following attributes:
+| miq_db_password    | `miq_app_password`  | The password of user specific in username used to connect to the database. |
+| miq_region         | 0                   | The ManageIQ region created in the database.                               |
+| miq_company        | My Company          | The company name of the appliance.                                         |
+| miq_disabled_roles | []                  | List of ManageIQ roles to disable on the appliance.                        |
```
maybe mentioning 'server roles' would be better? or somehow distinguish what roles are those

fabiendupont committed:
Remove redundant `default` filter.
Explicit the type of roles in README.
Add the list of default server roles to README.
